pkg/grpc/grpc.go

```diff
@@ -124,6 +124,10 @@ func (g *Manager) GetGRPCConnection(address, id string, namespace string, skipTL
 		return nil, err
 	}
 
+	if g.connectionPool[address] != nil {
+		g.connectionPool[address].Close()
+	}
+
```

Comment:
Can we add some unit-test for this case?

Reply:
I thought about it, but I couldn't figure out how I would detect the leak and fail the test?

Reply:
Yes it is hard to detect the leak. but I think that we can validate if connectionPool[address] is closed correctly and then have new value (new connection). :)

Reply:
Sure I can, but that happens regardless? Like, once we get to this point in the code, we get a new connection regardless of this change.

Reply:
Or maybe I'm missunderstanding, how will I validate that it has actually been closed?

Reply:
1. oldConn = Get old connection from connectionPool. 
2. Call GetGRPCConnection
3. oldConn.GetState() will return status of the connection. or you can try to oldConn.Close().Error() again  then you can check if it is closed or not.

Reply:
Ah, I follow you. Thanks for explaining.